gqltesting/testing.go

```diff
@@ -57,10 +58,22 @@ func RunTest(t *testing.T, test *Test) {
 	}
 
 	// Verify JSON to avoid red herring errors.
-	got, err := formatJSON(result.Data)
-	if err != nil {
-		t.Fatalf("got: invalid JSON: %s", err)
+	var got []byte
+
+	if test.RawResponse {
+		value, err := result.Data.MarshalJSON()
+		if err != nil {
+			t.Fatalf("got: unable to marshal JSON response: %s", err)
+		}
+		got = value
+	} else {
+		value, err := formatJSON(result.Data)
+		if err != nil {
+			t.Fatalf("got: invalid JSON: %s", err)
+		}
+		got = value
 	}
+
 	want, err := formatJSON([]byte(test.ExpectedResult))
```

Comment:
If raw response is used, do we need to leave the expected result raw as well?

Reply:
@eko ?

Reply:
Yes because the `RawResponse` only acts on `got` variable (effective response), not on the `want` (expected result) specified: in unit tests you have to specify a raw (not formatted) expected response.

Maybe the `formatJSON()` is not necessary anymore and we could just `json.Marshal()`